From bc153996a7b76fbaa7307356ab99e85e0924b6f7 Mon Sep 17 00:00:00 2001
From: Lu Xu <oliver_lew@outlook.com>
Date: Wed, 28 Aug 2024 13:28:57 +0800
Subject: [PATCH] fix: correct indent calculation for shiftwidth=0

Fixed the logic for calculating indentation in format.lua. Previously in
the case of shiftwidth=0, the indent was not using the value of tabstop,
leading to incorrect indentation.
---
 lua/copilot_cmp/format.lua | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/lua/copilot_cmp/format.lua b/lua/copilot_cmp/format.lua
index cd2478b..f527584 100644
--- a/lua/copilot_cmp/format.lua
+++ b/lua/copilot_cmp/format.lua
@@ -90,8 +90,12 @@ format.to_multi_line = function (item, ctx)
   local indent_level = indent_offset
   -- if spaces, force compliance with shiftwidth
   if user_indent == ' ' then
-    user_indent = string.rep(' ', vim.o.shiftwidth)
-    indent_level = math.floor(indent_offset/vim.o.shiftwidth)
+    local shiftwidth = vim.o.shiftwidth
+    if shiftwidth == 0 then
+      shiftwidth = vim.o.tabstop
+    end
+    user_indent = string.rep(' ', shiftwidth)
+    indent_level = math.floor(indent_offset/shiftwidth)
   end
 
   -- deindent everything and set all relative indents vim.o.shiftwidth spaces or one tab char
-- 
2.46.2

